cms/celery.py

```diff
@@ -10,11 +10,15 @@
 
 from celery import Celery
 
+from openedx.core.lib.celery import assert_celery_uninstantiated
 from openedx.core.lib.celery.routers import AlternateEnvironmentRouter
 
 # set the default Django settings module for the 'celery' program.
 os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'cms.envs.production')
 
+# Prevent instantiation of both CMS and LMS celery.
+# See assert_celery_uninstantiated for more details.
+assert_celery_uninstantiated('cms')
```

Comment:
In which process would this code be run?  In the CMS process or the CMS worker process?  I'm trying to determine:

1. which process would get killed.
2. does the mutex need to be updated at a later point.

Reply:
This would break if LMS or LMS Worker processes tried to import `cms.celery` directly.

It will not break if CMS or CMS Worker processes imported `cms.celery` multiple times.

Reply:
I believe it would be run in both. `cms/__init__.py` imports `cms.celery`, so if _any_ module under cms is loaded, this code runs. However, [Feanil's change](https://github.com/edx/edx-platform/commit/ab6bf348d4b0d7aec97598fb3118e076f2c096f4#diff-077fe64ca3f023f12cea200478dac5e52359f0fac037916735379b0acaf3355bR24) means that at least if the LMS process loads CMS code, this doesn't get loaded automatically. There's still the possibility that something imports it more directly, either now or in the future, and only in that case should this kill the process.

The assertion function should only get called once per process under normal conditions -- the `cms.celery` module is loaded for the first time, and the `SINGLETON_VARIANT` changes from `None` to `'cms'`. If the module is imported again, there's no additional call, since the module is already loaded.

The only time the function would be called multiple times is if both `lms.celery` and `cms.celery` are imported, which we very much want to avoid (and don't believe currently happens, after #25822).